File: php/features.php
<?php
declare(strict_types=1);

// ApicAgent SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ApicAgentFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ApicAgentBaseFeature();
            case "test":
                return new ApicAgentTestFeature();
            default:
                return new ApicAgentBaseFeature();
        }
    }
}

File: py/features.py
# ApicAgent SDK feature factory

from feature.base_feature import ApicAgentBaseFeature
from feature.test_feature import ApicAgentTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ApicAgentBaseFeature(),
        "test": lambda: ApicAgentTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
